### PR TITLE
extracted out sessionManager

### PR DESCRIPTION
**Pull request by DianeHu**
Extracted out sessionManager object, created in router file and passed at that point to necessary routes

TODO: figure out if streambuilder subscription to connection status in sessionManager should be used instead of returning futures

### CODE REVIEW

File: aux_ui/lib/screens/session_manager.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/services.dart';
```
Comment (j3remych3n) [inline aux_ui/lib/screens/session_manager.dart:1]: should prob move to lib or some folder for types / generic classes

File: aux_ui/lib/screens/session_manager.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import 'package:spotify_sdk/spotify_sdk.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:logger/logger.dart';
+
+class SessionManager {
+  var _authToken;
+  var _connected;
+  final Logger _logger = Logger();
+
+  void setStatus(String code, {String message = ""}) {
```
Comment (j3remych3n) [inline aux_ui/lib/screens/session_manager.dart:11]: what's this for

File: aux_ui/lib/screens/session_manager.dart (hunk 1 above)
Comment (DianeHu) [reply aux_ui/lib/screens/session_manager.dart]: moved

File: aux_ui/lib/screens/session_manager.dart (hunk 2 above)
Comment (DianeHu) [reply aux_ui/lib/screens/session_manager.dart]: debug

**Pull request merged**